tinyiiod-private.h

```diff
@@ -21,30 +21,30 @@
 #include "tinyiiod.h"
 
 char tinyiiod_read_char(struct tinyiiod *iiod);
-void tinyiiod_read(struct tinyiiod *iiod, char *buf, size_t len);
-ssize_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf, size_t len);
```

Comment:
removing the `, size_t len);` argument is a bad idea;
the `tinyiiod_read_line()` function won't know how large the `buf` memory is